node_modules/col-whiteboards/lib/api.js

```diff
@@ -1450,11 +1461,64 @@ var getWhiteboardAsPng = function(whiteboard, callback) {
       return callback({'code': 500, 'msg': 'Failed to convert the whiteboard to PNG'});
     }
 
-    return callback(null, Buffer.concat(pngChunks));
+    // The last chunk is the dimensions object
+    var dimensionsData = pngChunks.pop().toString('utf8');
+    try {
+      var dimensions = JSON.parse(dimensionsData)
+    } catch (err) {
+      return callback({'code': 500, 'msg': 'Failed to parse the dimensions data'});
+    }
+
+    return callback(null, Buffer.concat(pngChunks), dimensions);
   });
 };
 
 /**
+ * Generate a thumbnail from the full whiteboard image
+ *
+ * @param  {Whiteboard}   whiteboard                  The whiteboard for which a thumbnail is being generated
+ * @param  {String}       imagePath                   The path on disk to the full PNG representation of the whiteboard
+ * @param  {Function}     callback                    Standard callback function
+ * @param  {Object}       callback.err                An error object, if any
+ * @param  {String}       callback.thumbnailPath      The path on disk to the thumbnail
+ * @api private
+ */
+var generateThumbnailFromFullImage = function(whiteboard, imagePath, callback) {
+  var cmd = util.format('node ../data/resize.js %s', imagePath);
```

Comment:
Mostly because we now need a second`/whiteboard-preview-callback` endpoint to call back to and I thought it'd be more work to do this. I'll have a go at changing it.